pkgs/development/libraries/science/math/dealii/default.nix

```diff
@@ -0,0 +1,183 @@
+{ lib
+, stdenv
+, fetchzip
+, symlinkJoin
+, withDebug ? false
+, commonCxxFlags ? ""
+, debugCxxFlags ? ""
+, releaseCxxFlags ? ""
+, cmake
+, boost
+, withThreading ? true, tbb
+, withMPI ? true, mpi
+, with64bitIndices ? false
+, minimal ? false
+, doxygen
+, perl
+, graphviz
+, arpack
+, assimp
+, blas
+, lapack
+, gmsh
+, gsl
+, hdf5
+, hdf5-mpi
+, metis
+, muparser
+, p4est
+, p4est-sc
+, petsc
+, scalapack
+, symengine
+, zlib
+, suitesparse
+, openssh
+}:
+
+# Check PETSc compatibility
+assert withMPI -> petsc.mpiSupport;
+
+let
+  version = "9.4.0";
+
+  joinPkg = pkg: symlinkJoin {
```

Comment:
What is this about? Why are we not using the packages ad normal?

Reply:
Could you elaborate a bit? My intention was to include both normal and dev outputs of some packages.

Reply:
Is this required? Normally this is only required in special exceptions. What is the problem you encountered?

Also if this is really required please just add both packages next to each other.

Reply:
You were correct, I needed to specify `dev` output only for a single dependency.